### ISSUE TITLE
JsonSerializer.Serialize produces invalid JSON for [JsonExtensionData] property

### ISSUE THREAD

**Issue by KalininAndreyVictorovich** [dotnet/runtime#97225]
### Description

When serializing object containing property marked with `[JsonExtensionData]` attribute, serializer produces invalid JSON like this one: `{"Id":1,{"nested":true}}`

### Reproduction Steps

Run the following code

```C#
var mix = new Mix
{
    Id = 1,
    Extra = new() { ["nested"] = true, }
};

var text = System.Text.Json.JsonSerializer.Serialize(mix);
Console.WriteLine(text);
// output {"Id":1,{"nested":true}}


public class Mix
{
    public int Id { get; set; }

    [System.Text.Json.Serialization.JsonExtensionData]
    public System.Text.Json.Nodes.JsonObject? Extra { get; set; }
}

```

### Expected behavior

Correct JSON like `{"Id":1,"nested":true}` or at least valid JSON as if there was not `[JsonExtensionData]` attribute (`{"Id":1,"Extra":{"nested":true}}` ).

### Actual behavior

Invalid JSON `{"Id":1,{"nested":true}}`

### Regression?

Reproducible at least on .Net 6 to .Net 8

### Configuration

.Net 8
Windows 11
x64

**Comment by elgonzo** [comment]
There is a related question as to whether this scenario is supposed to be allowed or not. Because the [documentation for JsonExtensionDataAttribute](https://learn.microsoft.com/en-us/dotnet/api/system.text.json.serialization.jsonextensiondataattribute?view=net-8.0)states:

> The dictionary's TKey value **must be** [String](https://learn.microsoft.com/en-us/dotnet/api/system.string?view=net-8.0), and TValue must be [JsonElement](https://learn.microsoft.com/en-us/dotnet/api/system.text.json.jsonelement?view=net-8.0) or [Object](https://learn.microsoft.com/en-us/dotnet/api/system.object?view=net-8.0).

However, the example code in the report uses `JsonObject` as extension data property - an `IDictionary<string, JsonNode>`, which according to the JsonExtensionDataAttribute documentation is not an allowed dictionary type.

So, basically the problem is either
- the documentation is correct, but the (de)serializer fails to respond with an appropriate error/exception when encountering an unsupported extension data property type.
- `IDictionary<string, JsonNode>` is supposed to be supported as extension data property type, in which case there is both a bug in the implementation and an error in the `JsonExtensionDataAttribute` documentation.

**Comment by elgonzo** [comment]
> The documentation is correctly stating that `JsonObject` is one of the supported types. [...]

Where does the documentation for [JsonExtensionDataAttribute](https://learn.microsoft.com/en-us/dotnet/api/system.text.json.serialization.jsonextensiondataattribute?view=net-8.0) state this?

**Comment by eiriktsarpalis** [comment]
You're right, I misread the documentation which appears to be out of date. The correct statement on supported types can actually be found in the error messages of the implementation itself:

https://github.com/dotnet/runtime/blob/2d751cac7311b344c237df3b3c63b33434b52217/src/libraries/System.Text.Json/gen/Resources/Strings.resx#L150-L152

In other words, what I mentioned earlier holds. `JsonObject` _is_ supported and there is a bug specifically concerning serialization.

**Comment by elgonzo** [comment]
Okay. Regarding the documentation being out of date, should i file an issue report with the docs repo, or do you handle this internally between teams?

**Comment by eiriktsarpalis** [comment]
It would help if you could file a separate issue in dotnet-api-docs. Thanks!

**Comment by RutulPatel8** [comment]
> Can confirm that this occurs. It seems we never added testing for the serialization scenario -- deserialization appears to be working as expected.

Yes I have tested with Dot net core 6 version.
It is reproducible.
![image](https://github.com/dotnet/runtime/assets/84405086/ce08a3f1-e729-4613-9fd9-8690ee8b244a)

**Comment by TheArcaneBrony** [comment]
Still reproducible in dotnet 9
